lib/ansible/modules/windows/win_chocolatey_source.ps1

```diff
@@ -0,0 +1,411 @@
+    }
+
+    $sources = [System.Collections.ArrayList]@()
+    foreach ($xml_source in $choco_config.chocolatey.sources.GetEnumerator()) {
+        $source_username = $xml_source.Attributes.GetNamedItem("user")
+        if ($null -ne $source_username) {
+            $source_username = $source_username.Value
+        }
+        $source_password = $xml_source.Attributes.GetNamedItem("password")
+        if ($null -ne $source_password) {
+            $source_password = $source_password.Value
+        }
+
+        # 0.9.9.9+
+        $priority = $xml_source.Attributes.GetNamedItem("priority")
+        if ($null -ne $priority) {
+            $priority = [int]$priority.Value
+        }
+
+        # 0.9.10+
+        $certificate = $xml_source.Attributes.GetNamedItem("certificate")
+        if ($null -ne $certificate) {
+            $certificate = $certificate.Value
+        }
+        $certificate_password = $xml_source.Attributes.GetNamedItem("certificatePassword")
+        if ($null -ne $certificate_password) {
+            $certificate_password = $certificate_password.Value
+        }
+
+        # 0.10.4+
+        $bypass_proxy = $xml_source.Attributes.GetNamedItem("bypassProxy")
+        if ($null -ne $bypass_proxy) {
+            $bypass_proxy = [System.Convert]::ToBoolean($bypass_proxy.Value)
+        }
+        $allow_self_service = $xml_source.Attributes.GetNamedItem("selfService")
+        if ($null -ne $allow_self_service) {
+            $allow_self_service = [System.Convert]::ToBoolean($allow_self_service.Value)
+        }
+
+        # 0.10.8+
+        $admin_only = $xml_source.Attributes.GetNamedItem("adminOnly")
+        if ($null -ne $admin_only) {
+            $admin_only = [System.Convert]::ToBoolean($admin_only.Value)
+        }
+
+        $source_info = @{
+            name = $xml_source.id
+            source = $xml_source.value
+            disabled = [System.Convert]::ToBoolean($xml_source.disabled)
+            source_username = $source_username
+            priority = $priority
+            certificate = $certificate
+            bypass_proxy = $bypass_proxy
+            allow_self_service = $allow_self_service
+            admin_only = $admin_only
+
+            source_password = $source_password
+            certificate_password = $certificate_password
+        }
+        $sources.Add($source_info) > $null
+    }
+    return ,$sources
+}
+
+Function Set-RawChocolateySourceAttribute {
+    param(
+        $choco_app,
+        $name,
+        $attribute,
+        $value
+    )
+    # for private attributes like password/certificatePassword, the only way
+    # for us to preserve the existing value when changing another attirbute is
+    # to store it manually by editing the XML
+    $choco_config_path = "$(Split-Path -Path (Split-Path -Path $choco_app.Path))\config\chocolatey.config"
+    if (-not (Test-Path -Path $choco_config_path)) {
```

Comment:
When I see `-Path` in Powershell nowadays, I always wonder *Shouldn't this be `-LiteralPath` instead ?*

Reply:
Wouldn't hurt, probably isn't necessary as the path source is from `Get-Command` but will change it anyway.